test/unit/range/view/view_translate_join_test.cpp

```diff
@@ -0,0 +1,180 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include <seqan3/alphabet/nucleotide/all.hpp>
+#include <seqan3/alphabet/aminoacid/aa27.hpp>
+#include <seqan3/range/concept.hpp>
+#include <seqan3/range/view/char_to.hpp>
+#include <seqan3/range/view/complement.hpp>
+#include <seqan3/range/view/translate_join.hpp>
+#include <seqan3/std/ranges>
+
+#include "../iterator_test_template.hpp"
+
+using namespace seqan3;
+
+using iterator_type = decltype(view::translate_join(std::declval<std::vector<std::vector<seqan3::dna4> > &>()).begin());
+
+template <>
+struct iterator_fixture<iterator_type> : public ::testing::Test
+{
+    using iterator_tag = std::random_access_iterator_tag;
+    static constexpr bool const_iterable = true;
+
+    std::vector<std::vector<seqan3::dna4> > vec{"ACGTACGTACGTA"_dna4, "TCGAGAGCTTTAGC"_dna4};
+    std::vector<std::vector<aa27> > expected_range{{"TYVR"_aa27}, {"RTYV"_aa27}, {"VRT"_aa27}, {"YVRT"_aa27}, {"TYVR"_aa27}, {"RTY"_aa27},
+                                                   {"SRAL"_aa27}, {"REL*"_aa27}, {"ESFS"_aa27}, {"AKAL"_aa27}, {"LKLS"_aa27}, {"*SSR"_aa27}};
+    decltype(view::translate_join(vec)) test_range = view::translate_join(vec);
+
+    template <typename A, typename B>
+    static void expect_eq(A && begin_iterator_value, B && expected_range_value)
```

Comment:
```suggestion
    static void expect_eq(A && test_range_value, B && expected_range_value)
```
:nail_care: but not neccessary